app/routes.py

```diff
@@ -9,16 +9,27 @@
 #Home page
 @app.route('/', methods=['GET'])
 @app.route('/index', methods=['GET'])
-def index():
+@app.route('/index/<search_string>', methods=['GET'])
+def index(search_string=None):
     checkChallengesCompleted()
-    #Challenges
+    form = LoginForm()
+    search_form = SearchForm()
+    active_nav = "play"
     challengeList = []
-    challenges = Challenge.query.all()
+    if search_string is None:
+        challenges = Challenge.query.all()
+    else:
+        challenges = getChallengesByTitleOrCreator(search_string)
+        if challenges == []:
+            flash("Your search did not return any results.")
     for challenge in challenges:
         challengeList.append(challenge.toDict())
-    form = LoginForm()
-    active_nav = "play"
-    return render_template('index.html', challenges=challengeList, form=form, nav=active_nav)
+    return render_template('index.html', challenges=challengeList, form=form, search_form=search_form, nav=active_nav)
+
+@app.route('/search', methods=['POST'])
```

Comment:
You will need to update from main, and change @app.route to @main.route

Reply:
Done